File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/view/Fragment/AboutFragment.java
package com.byagowi.persiancalendar.view.Fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.byagowi.persiancalendar.R;
import com.byagowi.persiancalendar.Utils;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 * About Calendar Activity
 *
 * @author ebraminio
 */
public class AboutFragment extends Fragment {
    private final Utils utils = Utils.getInstance();

    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_about, container, false);

        String version = utils.programVersion(getContext());

        TextView versionTextView = (TextView) view.findViewById(R.id.version2);
        utils.prepareTextView(versionTextView);
        versionTextView.setText(getString(R.string.version) + " " +
                Utils.formatNumber(version, utils.preferredDigits(getContext())));

        StringBuilder sb = new StringBuilder();

        BufferedReader input = new BufferedReader(
                new InputStreamReader(getResources().openRawResource(R.raw.credits)));
        try {
            String line;
            while ((line = input.readLine()) != null) {
                sb.append(line);
                sb.append('\n');
            }
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            try {
                input.close();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }

        TextView licenseTextView = (TextView) view.findViewById(R.id.license);
        licenseTextView.setText("Android Persian Calendar Version "
                + version
                + "\n"
                + "Copyright (C) 2012-2016  ebrahim@gnu.org "
                + Utils.textShaper("ابراهیم بیاگوی")
                + "\n\n"
                + "This program is free software: you can redistribute it and/or modify "
                + "it under the terms of the GNU General Public License as published by "
                + "the Free Software Foundation, either version 3 of the License, or "
                + "(at your option) any later version.\n"
                + "\n"
                + "This program is distributed in the hope that it will be useful, "
                + "but WITHOUT ANY WARRANTY; without even the implied warranty of "
                + "MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the "
                + "GNU General Public License for more details.\n"
                + "\n"
                + "You should have received a copy of the GNU General Public License "
                + "along with this program.  If not, see http://www.gnu.org/licenses/.\n"
                + "\n"
                + "For bug report and credits: http://github.com/ebraminio/DroidPersianCalendar"
                + "\n\n----\n\n" + sb.toString());

        return view;
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/view/fragment/AboutFragment.kt
package com.byagowi.persiancalendar.view.fragment

import android.annotation.SuppressLint
import android.content.Intent
import android.content.pm.PackageManager
import android.graphics.Typeface
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.text.util.Linkify
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ScrollView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import com.byagowi.persiancalendar.R
import com.byagowi.persiancalendar.databinding.FragmentAboutBinding
import com.byagowi.persiancalendar.util.UIUtils
import com.byagowi.persiancalendar.util.Utils

/**
 * About Calendar Activity
 *
 * @author ebraminio
 */
class AboutFragment : Fragment() {

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
    val localActivity = activity ?: return null
    val binding = DataBindingUtil.inflate<FragmentAboutBinding>(inflater, R.layout.fragment_about,
        container, false)

    UIUtils.setActivityTitleAndSubtitle(localActivity, getString(R.string.about), "")

    // version
    val version = programVersion()
    binding.version.text = String.format(
        getString(R.string.version),
        Utils.formatNumber(version.split("-".toRegex())
            .dropLastWhile { it.isEmpty() }.toTypedArray()[0]))

    // licenses
    binding.licenses.setOnClickListener {
      val builder = AlertDialog.Builder(localActivity)
      builder.setTitle(resources.getString(R.string.about_license_title))
      val licenseTextView = TextView(localActivity)
      licenseTextView.text = Utils.readRawResource(localActivity, R.raw.credits)
      licenseTextView.setPadding(20, 20, 20, 20)
      licenseTextView.typeface = Typeface.MONOSPACE
      Linkify.addLinks(licenseTextView, Linkify.WEB_URLS or Linkify.EMAIL_ADDRESSES)
      val scrollView = ScrollView(localActivity)
      scrollView.addView(licenseTextView)
      builder.setView(scrollView)
      builder.setCancelable(true)
      builder.setNegativeButton(R.string.about_license_dialog_close) { _, _ -> }
      builder.show()
    }

    // help
    binding.aboutTitle.text = String.format(getString(R.string.about_help_subtitle),
        Utils.formatNumber(Utils.maxSupportedYear))

    // report bug
    binding.reportBug.setOnClickListener {
      try {
        startActivity(Intent(Intent.ACTION_VIEW,
            Uri.parse("https://github.com/ebraminio/DroidPersianCalendar/issues/new")))
      } catch (e: Exception) {
        e.printStackTrace()
      }
    }

    binding.email.setOnClickListener {
      val emailIntent = Intent(Intent.ACTION_SENDTO, Uri.fromParts("mailto", getString(R.string.about_mailto), null))
      emailIntent.putExtra(Intent.EXTRA_SUBJECT, getString(R.string.app_name))
      try {
        emailIntent.putExtra(Intent.EXTRA_TEXT, "\n\n\n\n\n\n\n" +
            "===Device Information===\nManufacturer: " + Build.MANUFACTURER + "\nModel: " + Build.MODEL + "\nAndroid Version: " + Build.VERSION.RELEASE + "\nApp Version Code: " + version.split("-".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[0])
        startActivity(Intent.createChooser(emailIntent, getString(R.string.about_sendMail)))
      } catch (ex: android.content.ActivityNotFoundException) {
        Toast.makeText(localActivity, getString(R.string.about_noClient), Toast.LENGTH_SHORT).show()
      }
    }

    return binding.root
  }

  private fun programVersion(): String {
    return try {
      context?.packageManager?.getPackageInfo(context?.packageName, 0)?.versionName ?: ""
    } catch (e: PackageManager.NameNotFoundException) {
      Log.e(AboutFragment::class.java.name, "Name not found on PersianCalendarUtils.programVersion")
      ""
    }
  }
}
